<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>RNAseq Differential Expressison — rna_diff_expr • rnaseq.work</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="RNAseq Differential Expressison — rna_diff_expr" />

<meta property="og:description" content="This function allows you to identify
differentially expressed genes." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rnaseq.work</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    How To
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/installation.html">Install</a>
    </li>
    <li>
      <a href="../articles/analyze.html">Obtain RNAseq Data for Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../articles/examples.html">Examples of Use</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../articles/faq.html">
    <span class="fa fa-question-circle-o"></span>
     
    FAQ
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-file-code-o"></span>
     
    Functions
  </a>
</li>
<li>
  <a href="https://github.com/homologus/rnaseq.work">
    <span class="fa fa-github fa-lg"></span>
     
    Github
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>RNAseq Differential Expressison</h1>
    
    <div class="hidden name"><code>rna_diff_expr.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function allows you to identify
differentially expressed genes.</p>
    
    </div>

    <pre class="usage"><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span> <span class='kw'>=</span> <span class='st'>"DESeq2"</span>)</pre>
        

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"DESeq"</span>)</div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'dplyr'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    filter, lag</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    intersect, setdiff, setequal, union</span></div><div class='output co'>#&gt; [1] "using DESeq"</div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocGenerics</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: parallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'BiocGenerics'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:parallel':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:dplyr':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    combine, intersect, setdiff, union</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:stats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,</span>
#&gt; <span class='message'>    colnames, colSums, dirname, do.call, duplicated, eval, evalq,</span>
#&gt; <span class='message'>    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,</span>
#&gt; <span class='message'>    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,</span>
#&gt; <span class='message'>    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,</span>
#&gt; <span class='message'>    rowSums, sapply, setdiff, sort, table, tapply, union, unique,</span>
#&gt; <span class='message'>    unsplit, which, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: Biobase</span></div><div class='output co'>#&gt; <span class='message'>Welcome to Bioconductor</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    Vignettes contain introductory material; view with</span>
#&gt; <span class='message'>    'browseVignettes()'. To cite Bioconductor, see</span>
#&gt; <span class='message'>    'citation("Biobase")', and for packages 'citation("pkgname")'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: locfit</span></div><div class='output co'>#&gt; <span class='message'>locfit 1.5-9.1 	 2013-03-22</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: lattice</span></div><div class='output co'>#&gt; <span class='message'>    Welcome to 'DESeq'. For improved performance, usability and</span>
#&gt; <span class='message'>    functionality, please consider migrating to 'DESeq2'.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: S4Vectors</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: stats4</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'S4Vectors'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:dplyr':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    first, rename</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: IRanges</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'IRanges'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:dplyr':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    collapse, desc, slice</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:grDevices':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    windows</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomicRanges</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: GenomeInfoDb</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: SummarizedExperiment</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: DelayedArray</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: matrixStats</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'matrixStats'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:Biobase':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyMissing, rowMedians</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:dplyr':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    count</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'DelayedArray'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:matrixStats':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:base':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    aperm, apply</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'DESeq2'</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from 'package:DESeq':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    estimateSizeFactorsForMatrix, getVarianceStabilizedData,</span>
#&gt; <span class='message'>    varianceStabilizingTransformation</span></div><div class='output co'>#&gt; <span class='error'>Error in ncol(countData) == nrow(colData): object 'count_table' not found</span></div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"DESeq2"</span>)</div><div class='output co'>#&gt; [1] "using DESeq2"</div><div class='output co'>#&gt; <span class='error'>Error in ncol(countData) == nrow(colData): object 'count_table' not found</span></div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"edgeR"</span>)</div><div class='output co'>#&gt; [1] "using edgeR"</div><div class='output co'>#&gt; <span class='message'>Loading required package: limma</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'limma'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:DESeq2':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:DESeq':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:BiocGenerics':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='output co'>#&gt; <span class='error'>Error in ncol(countData) == nrow(colData): object 'count_table' not found</span></div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"limma-voom"</span>)</div><div class='output co'>#&gt; [1] "using limma-voom"</div><div class='output co'>#&gt; <span class='error'>Error in ncol(countData) == nrow(colData): object 'count_table' not found</span></div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"sleuth"</span>)</div><div class='output co'>#&gt; [1] "using sleuth"</div><div class='output co'>#&gt; [1] 0</div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"baySeq"</span>)</div><div class='output co'>#&gt; [1] "using baySeq"</div><div class='output co'>#&gt; <span class='message'>Loading required package: abind</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: 'baySeq'</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from 'package:dplyr':</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    groups</span></div><div class='output co'>#&gt; <span class='message'>Finding priors...</span></div><div class='output co'>#&gt; <span class='message'>done.</span></div><div class='output co'>#&gt; <span class='message'>Length of priorReps:1000</span></div><div class='output co'>#&gt; <span class='message'>Length of priorSubset:1000</span></div><div class='output co'>#&gt; <span class='message'>Length of subset:1000</span></div><div class='output co'>#&gt; <span class='message'>Length of postRows:1000</span></div><div class='output co'>#&gt; ...</div><div class='output co'>#&gt;    annotation simA.1 simA.2 simA.3 simA.4 simA.5 simB.1 simB.2 simB.3 simB.4
#&gt; 1    count_80      1      1      0      1      1     13     21      8      6
#&gt; 2    count_78      1      1      0      1      1      8     13      7      9
#&gt; 3    count_66      0      0      0      0      0     15     10      4      4
#&gt; 4    count_21      2      0      1      1      0     15     15      6      5
#&gt; 5     count_7      9      8      8      8      9      1      2      1      0
#&gt; 6    count_26     13      4     11      5      7      1      1      1      0
#&gt; 7    count_64      6      6      8     11      9      1      1      0      0
#&gt; 8    count_72      0      0      1      0      0      7      6      4      3
#&gt; 9    count_83     14      6      9      2      9      1      0      0      1
#&gt; 10   count_27      5      3      6      4      7      0      0      0      1
#&gt;    simB.5 Likelihood ordering       FDR.DE      FWER.DE
#&gt; 1      20  0.9995791      2&gt;1 0.0004208781 0.0004208781
#&gt; 2      10  0.9992829      2&gt;1 0.0005689902 0.0011376785
#&gt; 3      10  0.9990943      2&gt;1 0.0006812318 0.0020423631
#&gt; 4      11  0.9985733      2&gt;1 0.0008675879 0.0034661055
#&gt; 5       0  0.9983926      1&gt;2 0.0010155560 0.0050679624
#&gt; 6       0  0.9980266      1&gt;2 0.0011752048 0.0070314101
#&gt; 7       1  0.9962283      1&gt;2 0.0015461322 0.0107765864
#&gt; 8       8  0.9957691      2&gt;1 0.0018817294 0.0149619013
#&gt; 9       1  0.9954451      1&gt;2 0.0021787487 0.0194486539
#&gt; 10      0  0.9941751      1&gt;2 0.0025433681 0.0251603097</div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"NOISeq"</span>)</div><div class='output co'>#&gt; [1] "using NOISeq"</div><div class='output co'>#&gt; [1] 0</div><div class='input'><span class='fu'>rna_diff_expr</span>(<span class='no'>count_table</span>, <span class='no'>design_table</span>, <span class='kw'>method</span><span class='kw'>=</span><span class='st'>"EBSeq"</span>)</div><div class='output co'>#&gt; [1] "using EBSeq"</div><div class='output co'>#&gt; [1] 0</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
            
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Manoj Samanta.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

